File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPRemainingAir.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPRemainingAir implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "remaining_air";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 360;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getRemainingAir();
    }

    @Override
    public void restoreProperty(Player player, Integer ticks) {
        player.setRemainingAir(ticks);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPRemainingAir.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPRemainingAir extends PlayerStateProperty[Integer]:
  override def getConfigName: String = "remaining_air"

  override def getDefaultSerializedProperty: Integer = 360

  override def getSerializedPropertyFromPlayer(player: Player): Integer =
    player.getRemainingAir

  override def restoreProperty(player: Player, ticks: Integer): Unit =
    player.setRemainingAir(ticks)